lms/services/email_preferences.py

```diff
@@ -16,12 +18,37 @@ class InvalidTokenError(Exception):
     pass
 
 
+@dataclass(frozen=True)
+class EmailPrefs:  # pylint:disable=too-many-instance-attributes
+    DAYS = ["mon", "tue", "wed", "thu", "fri", "sat", "sun"]
+
+    h_userid: str
+    mon: bool = field(default=True)
+    tue: bool = field(default=True)
+    wed: bool = field(default=True)
+    thu: bool = field(default=True)
+    fri: bool = field(default=True)
+    sat: bool = field(default=True)
+    sun: bool = field(default=True)
+
+    def days(self) -> dict:
+        return {key: value for key, value in asdict(self).items() if key in self.DAYS}
```

Comment:
The `frozen=True` is good because any mutations to an `EmailPrefs` object returned by `EmailPreferencesService.get_preferences()` would not be automatically saved to the DB. This forces you to create a new `EmailPrefs` object and pass it into `EmailPreferencesService.set_preferences()`.